frontend/src/store/store.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { createContext, useReducer, useEffect } from 'react';
+
+// Default state for the store
+const initialStore = {
+  sessionID: '', // Empty string represents user not authenticated, otherwise they are authenticated
+  plan: '', // "FREE" or "ENTERPRISE"
+};
+
+// On app load, try to fetch a saved store from localStorage
+const localStore = JSON.parse(localStorage.getItem('store'));
+
+// Updates the store with newStoreValues
+// Pass `newStoreValues = null` to reset the store
+const storeReducer = (store, newStoreValues) => {
+  if (newStoreValues === null) {
+    localStorage.removeItem('store');
+    return initialStore;
+  }
+  return { ...store, ...newStoreValues };
+};
+
+// Global React.Context value for components to pass into useContext in order to access the store
+const StoreContext = createContext();
+
+// createStore should wrap the entire application, so that any component in the application
+// can access the global store's state as a StoreContext.Consumer
+const createStore = WrappedComponent => props => {
```

Comment:
instead of HOC I would use a regular component as it's easier to read and follow, especially since we need only 1 context provider per application.

```ts
const AppContext = props => {
  // 
}
```
and then

```tsx
 <AppContext>
    <Router>
      <Switch>
        <Route exact path="/">
          <Login />
        </Route>
        <Route path="/dashboard">
          <Dashboard />
        </Route>
      </Switch>
    </Router>
</AppContext>
```

Reply:
Good call, this does look a lot nicer